tools/clang/include/clang/SPIRV/SpirvBuilder.h

```diff
@@ -666,6 +671,48 @@ class SpirvBuilder {
       SpirvInstruction *constOffsets, SpirvInstruction *sample,
       SpirvInstruction *minLod);
 
+  /// \brief Creates an access chain instruction to retrieve the element from
```

Comment:
Why do we need all of these specialized functions? Why can't we call the existing SpirvBuilder methods from SpirvEmitter's entry function wrapper?

Reply:
I updated the `SpirvBuilder` to switch `insertPoint` to `moduleInitInsertPoint` and use the existing methods to create instructions, but the SpirvEmitter's entry function wrapper does not call those methods. As I mentioned in the comment about handling CTBuffer decls in the SpirvEmitter's entry function wrapper, I do not think we can do it there because decls are handled by `doDecl`.

Reply:
We had an offline discussion. Let me mark it as "resolved".